src/main/java/core/basesyntax/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+    private final Storage storage;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy, Storage storage) {
+        this.operationStrategy = operationStrategy;
+        this.storage = storage;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+            OperationHandler handler = operationStrategy.getHandler(transaction.getOperation());
+            if (handler != null) {
+                handler.handleOperation(transaction, storage);
```

Comment:
Directly passing the `Storage` to the `handleOperation` method violates the encapsulation principle. The `OperationHandler` should not directly interact with `Storage` but rather return a result that the `ShopService` can use to update `Storage`.